File: ChatApp/Common/ActionResults/RequestResult.cs
using Microsoft.AspNetCore.Mvc;

namespace ChatApp.Common.ActionResults
{
    public class RequestResult : OkObjectResult
    {
        private const int DefaultStatusCode = ResponseStatusCodes.RequestResult51;
        
        public RequestResult(object value) : base(value)
        {
           // StatusCode = DefaultStatusCode;
            IsSuccess = true;
        }
        
        public RequestResult(string errorMessage) 
            : base(null)
        {
            //StatusCode = DefaultStatusCode;
            IsSuccess = false;
            ErrorMessage = errorMessage;
        }

        public bool IsSuccess { get; }
        
        public string ErrorMessage { get; set; }
    }
}

File: ChatApp/client-app/src/common/RequestResult.ts
export interface RequestOkResult<T> {
    value: T,
    errorMessage: null
    hasValue: true
}

export interface RequestBadResult<T> {
    value: null,
    errorMessage: string,
    hasValue: false
}
export type RequestResult<T> = RequestBadResult<T> | RequestOkResult<T>